client/order/index.js

```diff
@@ -97,4 +108,110 @@ jQuery( function ( $ ) {
 		document.body.appendChild( container );
 		ReactDOM.render( modalToRender, container );
 	}
+
+	function maybeShowDisputeNotice( disputeData ) {
+		if ( ! disputeData ) {
+			return;
+		}
+
+		const container = document.querySelector(
+			'#wcpay-order-payment-details-container'
+		);
+
+		// If the container doesn't exist (WC < 7.9), return.
+		if ( ! container ) {
+			return;
+		}
+
+		const now = moment();
+		const dueBy = moment.unix( disputeData.dueBy );
+		const countdownDays = Math.floor( dueBy.diff( now, 'days', true ) );
+
+		// If the dispute is due in the past, we don't want to show the notice.
+		if ( now.isAfter( dueBy ) ) {
+			return;
+		}
+
+		const amountFormatted = formatExplicitCurrency(
+			disputeData.amount,
+			disputeData.currency
+		);
+
+		let urgency = 'warning';
+		let buttonLabel = __( 'Respond now', 'woocommerce-payments' );
+		let title = sprintf(
+			// Translators: %1$s is the formatted dispute amount, %2$s is the dispute reason, %3$s is the due date.
+			__(
+				'This order has a chargeback dispute of %1$s labeled as "%2$s". Please respond to this dispute before %3$s.',
+				'woocommerce-payments'
+			),
+			amountFormatted,
+			reasons[ disputeData.reason ].display,
+			dateI18n( 'M j, Y', dueBy.local().toISOString() )
+		);
+		let suffix = '';
+
+		// If the dispute is due within 7 days, use different wording.
+		if ( 7 > countdownDays ) {
+			title = sprintf(
+				// Translators: %1$s is the formatted dispute amount, %2$s is the dispute reason, %3$s is the due date.
+				__(
+					'Please resolve the dispute on this order for %1$s labeled "%2$s" by %3$s.',
+					'woocommerce-payments'
+				),
+				amountFormatted,
+				reasons[ disputeData.reason ].display,
+				dateI18n( 'M j, Y', dueBy.local().toISOString() )
+			);
+			suffix = sprintf(
+				// Translators: %d is the number of days left to respond to the dispute.
+				_n(
+					'(%s day left)',
+					'(%s days left)',
+					countdownDays,
+					'woocommerce-payments'
+				),
+				countdownDays
+			);
+		}
+
+		// If the dispute is due within 72 hours, we want to highlight it as urgent/red.
+		if ( 3 > countdownDays ) {
+			urgency = 'error';
+		}
+
+		if ( 1 > countdownDays ) {
```

Comment:
```suggestion
		if ( 1 >= countdownDays ) {
```

Reply:
This is using `>` since we only want this to resolve `true` if within 24h.
24+ hours would result in `countdownDays = 1` after `Math.floor` which we want to resolve as `false` (not within 24h).

E.g. 23 hours = 0.95 days = `Math.floor(0.95) = 0 countdownDays`.

This can be tested by setting `disputeData.dueBy = moment().add( 23, 'hours' ).unix();`